src/Microsoft.ApplicationInsights/TelemetryClientExtensions.cs

```diff
@@ -132,12 +159,31 @@ public static IOperationHolder<T> StartOperation<T>(this TelemetryClient telemet
 
                 if (parentActivity == null)
                 {
-                    // telemetryContext.Id is always set: if it was null, it is set to opTelemetry.Id and opTelemetry.Id is never null
-                    operationActivity.SetParentId(telemetryContext.Id);
+                    // telemetryContext.Id is always set: if it was null, it is set to newly generated W3C compatible TraceID
+                    if (Activity.DefaultIdFormat == ActivityIdFormat.W3C)
+                    {
+                        // There is no need of checking if TelemetryContext.ID is W3C Compatible. It is always set to 
+                        // W3C compatible id. Even user supplied non-compatible ID is ignored.                                                
+                        operationActivity.SetParentId(string.Join("-", W3CConstants.DefaultVersion, telemetryContext.Id, W3CConstants.InvalidSpanID, W3CConstants.DefaultTraceFlag));
```

Comment:
still SetParentId(ActivityTraceId.CreateFromString(telemetryContext.Id.AsSpan()), default, ActivityTraceFlags.None) is cleaner and does not throw, please use that one